addons/print/models/res_users.py

```diff
@@ -11,25 +11,26 @@ class User(models.Model):
     _inherit = 'res.users'
 
     printer_ids = fields.Many2many('print.printer', string="Default Printers")
-    printer_id = fields.Many2one('print.printer', string="Default Printer",
-                                 compute='_compute_printer_id', store=True)
 
-    @api.multi
-    @api.depends('printer_ids')
-    def _compute_printer_id(self):
-        """Compute default ungrouped printer (for backwards compatibility)"""
-        for user in self:
-            user.printer_id = user.printer_ids.filtered(
-                lambda x: not x.group_id
-            )
+    def _get_printer(self, report_type=None):
+        """
+        Identify and return user default ungrouped printer (for backwards compatibility) 
+        for report type (if specified)
+        """
+        self.ensure_one()
+        printer = self.printer_ids.filtered(
+            lambda p: not p.group_id and (report_type is None or p.report_type == report_type)
```

Comment:
similar to the previous question, though this is different logic, I'm sure I'm missing something

Reply:
In this case we are saying get the default printer for the user in `self`. If `report_type` is set to `qweb-pdf` for example, it will return the default PDF printer for the user, if there is one.

However if `report_type` is `None`, we don't want to filter out printers based on `report_type`, as no printer would be included.

It probably doesn't make sense to call `_get_printer` without `report_type` being set, but I kept this in so that it works as close as possible to the removed `printer_id` computed field.